src/gdrapi.c

```diff
@@ -537,59 +537,70 @@ static inline int ptr_is_aligned(const void *ptr, unsigned powof2)
 
 static int gdr_copy_to_mapping_internal(void *map_d_ptr, const void *h_ptr, size_t size, int wc_mapping)
 {
-    if (first_time) {
-        gdr_init_cpu_flags();
-    }
-
     do {
+        // For very small sizes and aligned pointers, we use simple store.
+        if (unlikely(size == 0)) {
```

Comment:
can you move this check one level up, i.e. in gdr_copy_to_mapping()

Reply:
Done.